### ISSUE TITLE
persistence,s3: "error: location "pstoev-persistence-test/LOCK" was already_locked" after double restart

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#10225]
### What version of Materialize are you using?

v0.18.1-dev (fc13fcb9f)

### How did you install Materialize?

Built from source

### What was the issue?

If the persistence test is patched as follows to use s3:

```
diff --git a/test/persistence/mzcompose.py b/test/persistence/mzcompose.py
index 40e22e28f..54561b654 100644
--- a/test/persistence/mzcompose.py
+++ b/test/persistence/mzcompose.py
@@ -31,6 +31,10 @@ mz_disable_user_indexes = Materialized(
     options=f"{mz_options} --disable-user-indexes",
 )
 
+mz_s3 = Materialized(
+    options=f"{mz_options} --persist-storage-enabled --persist-storage=s3://mtlz-test-persist-1d-lifecycle-delete/pstoev-persistence-test"
+)
+
 prerequisites = ["zookeeper", "kafka", "schema-registry"]
 
 SERVICES = [
@@ -45,7 +49,7 @@ td_test = os.environ.pop("TD_TEST", "*")
 
 
 def workflow_persistence(c: Composition) -> None:
-    for mz in [mz_default, mz_logical_compaction_window_off]:
+    for mz in [mz_s3, mz_default, mz_logical_compaction_window_off]:
         with c.override(mz):
             workflow_kafka_sources(c)
             workflow_user_tables(c)
```

Running it will cause Mz to fail post-restart:

```
materialized: persistence error: location "pstoev-persistence-test/LOCK" was already_locked:
fd40219b-5814-47ac-9a4e-b162ea78beac
materialized v0.18.1-dev (fc13fcb9f)
os: Ubuntu 20.4.0 [64-bit]
start time: 2022-01-25 11:39:23.513243968 UTC
num workers: 2
```

### Relevant log output

```source
2022-01-25T11:39:27.923317Z  INFO imds_load_region: aws_config::imds::region: loaded region from IMDS region=us-east-2
2022-01-25T11:39:27.948279Z  INFO send_operation{operation="GetObject" service="s3"}: aws_config::meta::credentials::chain: provider in chain did not provide credentials provider=Environment context=environment variable not set
2022-01-25T11:39:27.948337Z  INFO send_operation{operation="GetObject" service="s3"}: aws_config::meta::credentials::chain: provider in chain did not provide credentials provider=Profile context=No profiles were defined
2022-01-25T11:39:27.948347Z  INFO send_operation{operation="GetObject" service="s3"}: aws_config::meta::credentials::chain: provider in chain did not provide credentials provider=WebIdentityToken context=$AWS_WEB_IDENTITY_TOKEN_FILE was not set
2022-01-25T11:39:27.948357Z  INFO send_operation{operation="GetObject" service="s3"}: aws_config::meta::credentials::chain: provider in chain did not provide credentials provider=EcsContainer context=ECS provider not configured
2022-01-25T11:39:27.956293Z  INFO send_operation{operation="GetObject" service="s3"}: aws_config::meta::credentials::chain: loaded credentials provider=Ec2InstanceMetadata
materialized: persistence error: location "pstoev-persistence-test/LOCK" was already_locked:
fd40219b-5814-47ac-9a4e-b162ea78beac
materialized v0.18.1-dev (fc13fcb9f)
os: Ubuntu 20.4.0 [64-bit]
start time: 2022-01-25 11:39:23.513243968 UTC
num workers: 2
```

**Comment by philip-stoev** [comment]
This seems to be caused by the rapid two Mz restarts in succession in the middle of the test. If Mz is restarted only once, the test is able to complete.

**Comment by danhhz** [comment]
I haven't looked at what the test is doing in detail yet, but just double checking, are there two different catalog ids (i.e. unique mzdata dirs) involved here? You're hardcoding a single storage location (`s3://mtlz-test-persist-1d-lifecycle-delete/pstoev-persistence-test`) and this would be the expected behavior if you tried to point a different materialize at it.

**Comment by danhhz** [comment]
No longer relevant in new persist, there are no lock files, so this should be working now.
